File: client-app/src/app/models/order.ts
import { PortionOrder } from "./portionOrder";
import { User } from "./user";

export interface Order {
    id?: string; 
    date?: Date;
    user?: User;
    portions: PortionOrder[];
}

export class Order implements Order {
    id?: string = undefined; 
    date?: Date;
    user?: User;
    portions: PortionOrder[];

    constructor(portions: PortionOrder[]) {
        this.portions = portions;
    }
}

File: Domain/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Order : BaseEntity<Guid>
    {
        public User User { get; set; }
        public ICollection<Portion> Portions { get; set; } = new List<Portion>();
        public DateTime Date { get; set; } = DateTime.UtcNow;
    }
}